src/Model/FtpUploader.php

```diff
@@ -22,6 +22,26 @@ public function __construct(FtpConfig $config, IoInterface $client)
         $this->client = $client;
     }
 
+    /**
+     * @param array $params
+     * @throws \Exception When credentials are invalid or target directory is not writeable
+     */
+    public function testConnection(array $params): void
+    {
+        $fileName = 'testconnection';
+        $this->client->open($params);
+        //check write permission
+        $this->client->write($fileName, '');
+        $this->client->rm($fileName);
+
+        $this->client->close();
```

Comment:
Can we pack the last 3 lines in a try/finally block? Otherwise the connection stays open.

Reply:
changed